src/main/scala/mesosphere/marathon/core/task/termination/impl/TaskKillServiceActor.scala

```diff
@@ -0,0 +1,181 @@
+package mesosphere.marathon.core.task.termination.impl
+
+import akka.actor.{ Actor, ActorLogging, Cancellable, Props }
+import mesosphere.marathon.MarathonSchedulerDriverHolder
+import mesosphere.marathon.core.base.Clock
+import mesosphere.marathon.core.task.termination.TaskKillConfig
+import mesosphere.marathon.state.Timestamp
+import mesosphere.marathon.core.task.{ Task, TaskStateOp }
+import mesosphere.marathon.core.task.tracker.{ TaskStateOpProcessor, TaskTracker }
+import mesosphere.marathon.core.event.MesosStatusUpdateEvent
+
+import scala.collection.mutable
+import scala.concurrent.Promise
+
+/**
+  * An actor that handles killing tasks in chunks and depending on the task state.
+  * Lost tasks will simply be expunged from state, while active tasks will be killed
+  * via the scheduler driver. There is be a maximum number of kills in flight, and
+  * the service will only issue more kills when tasks are reported terminal.
+  */
+private[impl] class TaskKillServiceActor(
```

Comment:
(a) what do you think about modeling this as an FSM actor?
(b) what do you think about batching together kills that arrive within some window (so that back-to-back individual kills don't have to wait for the retry period)?